ambari/src/main/java/io/brooklyn/ambari/AmbariClusterImpl.java

```diff
@@ -79,47 +62,39 @@
+        setAttribute(AMBARI_SERVER, addChild(createServerSpec(getConfig(SECURITY_GROUP))));
+
+        List<String> services = getConfig(HADOOP_SERVICES);
+        if (isServicesBasedDeployment(services)) {
+            createSingleClusterOfAgents();
+        } else {
+            calculateTotalAgentsInHostgroups();
+        }
 
-        Object securityGroup = getConfig(SECURITY_GROUP);
-        setAttribute(AMBARI_SERVER, addChild(createServerSpec(hostnameSensor, securityGroup)));
-        
-        setAttribute(AMBARI_AGENTS, addChild(EntitySpec.create(DynamicCluster.class)
-            .configure(DynamicCluster.INITIAL_SIZE, getRequiredConfig(INITIAL_SIZE))
-            .configure(DynamicCluster.MEMBER_SPEC, createAgentSpec(hostnameSensor, securityGroup))
-            .displayName("All Nodes")
-        ));
-        
         addEnricher(Enrichers.builder()
-            .propagating(Attributes.MAIN_URI)
-            .from(getAttribute(AMBARI_SERVER))
-            .build());
+                .propagating(Attributes.MAIN_URI)
+                .from(getAttribute(AMBARI_SERVER))
+                .build());
     }
 
-    private EntitySpec<? extends AmbariServer> createServerSpec(SshCommandSensor<String> hostnameSensor, Object securityGroup) {
+    private EntitySpec<? extends AmbariServer> createServerSpec(Object securityGroup) {
         EntitySpec<? extends AmbariServer> serverSpec = getConfig(SERVER_SPEC)
-            .configure(SoftwareProcess.SUGGESTED_VERSION, getConfig(AmbariCluster.SUGGESTED_VERSION))
-            .displayName("Ambari Server")
-            .addInitializer(hostnameSensor);
-        if(securityGroup != null){
+                .configure(SoftwareProcess.SUGGESTED_VERSION, getConfig(AmbariCluster.SUGGESTED_VERSION))
+                .displayName("Ambari Server");
+        if (securityGroup != null) {
             serverSpec.configure(SoftwareProcess.PROVISIONING_PROPERTIES.subKey("securityGroups"), securityGroup);
         }
         return serverSpec;
     }
-    
 
-    private EntitySpec<? extends AmbariAgent> createAgentSpec(SshCommandSensor<String> hostnameSensor, Object securityGroup) {
+    private EntitySpec<? extends AmbariAgent> createAgentSpec(Object securityGroup) {
         EntitySpec<? extends AmbariAgent> agentSpec = getConfig(AGENT_SPEC)
-            .configure(AmbariAgent.AMBARI_SERVER_FQDN, 
-                attributeWhenReady(getAttribute(AMBARI_SERVER),AmbariServer.HOSTNAME))
-            .configure(SoftwareProcess.SUGGESTED_VERSION, 
-                getConfig(AmbariCluster.SUGGESTED_VERSION))
-            //TODO shouldn't use default os
-            .addInitializer(hostnameSensor);
-        if(securityGroup != null){
+                .configure(AmbariAgent.AMBARI_SERVER_FQDN,
+                        attributeWhenReady(getAttribute(AMBARI_SERVER), AmbariServer.HOSTNAME))
+                .configure(SoftwareProcess.SUGGESTED_VERSION,
+                        getConfig(AmbariCluster.SUGGESTED_VERSION));
```

Comment:
This commit is just moving around some existing code.  I suggest we review and fix in a separate pr.